reference/3.0/Microsoft.PowerShell.Core/About/about_Trap.md

```diff
@@ -242,58 +249,87 @@ At C:\PS>TestScript2.ps1:4 char:19
-the Break keyword in a Trap statement:
+If you include a `Break` statement in a `Trap` statement list, PowerShell
+stops the function or script. The following sample function uses the `Break`
+keyword in a `Trap` statement:
 
 ```powershell
-C:\PS> function break_example {
-    trap {"Error trapped"; break;}
+function break_example {
+    trap {
+        "Error trapped"
+        break
+    }
     1/$null
     "Function completed."
-    }
+}
+
+break_example
+```
 
-C:\PS> break_example
+```Output
 Error trapped
 Attempted to divide by zero.
 At line:4 char:7
 ```
 
-Because the Trap statement included the Break keyword, the function does
+Because the `Trap` statement included the `Break` keyword, the function does
 not continue to run, and the "Function completed" line is not run.
 
-If you include a Continue statement in a Trap statement, Windows PowerShell
-resumes after the statement that caused the error, just as it would without
-Break or Continue. With the Continue keyword, however, Windows PowerShell
-does not write an error to the error stream.
+If you include a `Continue` statement in a `Trap` statement, PowerShell resumes
+after the statement that caused the error, just as it would without `Break` or
+`Continue`. With the `Continue` keyword, however, PowerShell does not write an
+error to the error stream.
 
-The following sample function uses the Continue keyword in a Trap
+The following sample function uses the `Continue` keyword in a `Trap`
 statement:
 
 ```powershell
-C:\PS> function continue_example {
-    trap {"Error trapped"; continue;}
+function continue_example {
+    trap {
+        "Error trapped"
+        continue
+    }
     1/$null
-    "Function completed."}
+    "Function completed."
+}
+
+continue_example
+```
 
-C:\PS> continue_example
+```Output
 Error trapped
 Function completed.
 ```
 
-The function resumes after the error is trapped, and the "Function
-completed" statement runs. No error is written to the error stream.
+The function resumes after the error is trapped, and the "Function completed"
+statement runs. No error is written to the error stream.
+
+> [!IMPORTANT]
+> `Trap` statements can appear anywhere within the scope your code. When
+> PowerShell parses the code, the `Trap` definitions within a scope are
+> collected and *hoisted* to the top of that scope. This means that traps can
+> be defined in any order and at any time within a function or script. All
+> defined traps are available at runtime even if execution has not advanced to
+> the point where the traps was defined in the source file. When a terminating
+> error occurs, PowerShell searches for the trap with the most specific match,
+> starting in the current scope of execution.
+>
+> `Trap` statements are useful for post-mortem error analysis but are not good
+> at error handling. For robust error handling, use `try/catch` blocks. Traps
+> are defined using `Catch` statements, but those traps only apply to the code
```

Comment:
This section isn't really true; `trap` and `try`/`catch` are related conceptually and to some extent in the implementation, but we define the semantics of each independently and are free to change the underlying implementation.

Reply:
I removed the opinion statement about fitness for error handling. I had copied that from your comments in the original issue discussion. Are there other parts of this that are untrue or need further clarification? I think it is worth mentioning `try`/`catch` as an alternative to `trap`.